src/pint/models/parameter.py

```diff
@@ -583,8 +583,10 @@ def add_alias(self, alias):
 
     def name_matches(self, name):
         """Whether or not the parameter name matches the provided name"""
-        return (name == self.name.upper()) or (
-            name in [x.upper() for x in self.aliases]
+        return (
+            (name == self.name.upper())
+            or (name in [x.upper() for x in self.aliases])
+            or (name.replace("0", "") == self.name.upper())
```

Comment:
Isn't this going to reduce `DM10` to `DM1`?

Reply:
This was a very silly thing for me to do in retrospect, I didn't even implement the bad workaround as I meant to. I think I have something better though. Just working on the tests.